enterprise/core-edge/src/main/java/org/neo4j/coreedge/raft/RaftInstance.java

```diff
@@ -226,13 +278,14 @@ protected void handleOutcome( Outcome<MEMBER> outcome ) throws RaftStorageExcept
 
             logShipping.handleCommands( outcome.getShipCommands(), leaderContext );
         }
-        else if ( oldLeader != null && oldLeader.equals( myself ) && !outcome.getLeader().equals( myself ) ) // TODO: inspect the reported issue
+        else if ( myself.equals( oldLeader ) && !myself.equals( outcome.getLeader() ) )
         {
             logShipping.stop();
         }
 
         // Update state
         state.update( outcome );
+        notifyAll();
```

Comment:
It is a bit strange to have `notifyAll()` in a protected method because it could be overridden and thus `notifyAll()` might never be called. At the same time method is not synchronized but could be called from the outside leading to `IllegalMonitorStateException`s.